Scenarios/BlueGreen-Deployment-for-AKS/GAP/bluegreen-aks-solution-content.md

```diff
@@ -0,0 +1,222 @@
+4. T3: Traffic Switch to the green cluster
+5. T4: Blue cluster is destroyed
+
+Once implemented, the workflow will start at the next planned release of the cluster beginning with T1: Green Cluster Deployment. The triggers to transition in the multiples stages can be programtically configured and executed, this is usually the desired end state, that start from a manual/semi-automatic implementation. The triggers are related to functional parameters together with SLO/SLI defined at operation levele to cover apps and infra aspects.
+This pattern is flexible on the network discoverabiity of the clusters, in fact you can have multiple options:
+- A DNS record dedicated to the blue and green clusters IP
+- A DNS record dedicated to the blue green cluster pointing to the App Gateway IP
+
+
+#### T0: Blue Cluster is On
+The initial stage of the pattern is to have the existing live cluster on, which is called the Blue Cluster. At this stage we are preparing for the deployment of the new version of the cluster.
+
+![Step0](../media/bg-step0.png)
+
+The trigger condition for the [next stage](#t1-green-cluster-deployment) is the release of a new version of the cluster.
+
+
+#### T1: Green Cluster Deployment
+Now the deployment of the new version is started and the the new Green Cluster is deployed in parallel to the existing Blue Cluster. After the Green cluster is deployed, the live traffic is still routed in the Blue Cluster. 
+
+![Step1](../media/bg-step1.png)
+
+The trigger to move into the [T2 stage](#t2-sync-k8s-state-between-blue-and-green-cluster) is the validation of AKS at platform level, using native metrics available in Azure Monitor and CLI commands to check the health of the deployment.
+
+
+#### T2: Sync K8S State between Blue and Green cluster
+At this stage there is an parity between the two clusters, meaning all **applications**, **operators** and **K8S resources** are newly deployed in the green cluster. The ultimate goal is that at the end of the sync the clusters are equivalent.
+
+There are multiple solutions/approaches to replicate/sync K8S state on clusters:
+- Redeployment via CI/CD
+- GitOps with solutions promoted in CNCF
+- Customized solution that store the K8S configs and resources, usually databases and K8S manifests generators
+
+![Step2](../media/bg-step2.png)
+
+Usually syncing the deployment of new applications is not permitted in the live cluster, so there is a period of time that starts with the sync and finishs when the switch to the green is completed. This period can be avoided with advanced mechanisms that manage the K8S state in multiple clusters, but that is not part of this article.
+
+When the sync is completed, a test/validation of the cluster is required. This include a check on the monitoring and logging platforms to validate the health of the cluster. Usually the Green cluster is exposed on the App Gateway or External LB with an internal URL, that is not visibile for external users.
```

Comment:
Change "This include" to "This includes".

Reply:
fixed